pandas/tests/indexing/test_loc.py

```diff
@@ -587,17 +587,36 @@ def test_loc_modify_datetime(self):
 
         tm.assert_frame_equal(df, expected)
 
-    def test_loc_setitem_frame_with_reindex(self):
+    def test_loc_setitem_frame_with_reindex(self, using_array_manager):
         # GH#6254 setting issue
         df = DataFrame(index=[3, 5, 4], columns=["A"], dtype=float)
         df.loc[[4, 3, 5], "A"] = np.array([1, 2, 3], dtype="int64")
 
         # setting integer values into a float dataframe with loc is inplace,
         #  so we retain float dtype
         ser = Series([2, 3, 1], index=[3, 5, 4], dtype=float)
+        if using_array_manager:
+            # TODO(ArrayManager) with "split" path, we still overwrite the column
+            # and therefore don't take the order of the indexer into account
+            ser = Series([1, 2, 3], index=[3, 5, 4], dtype="int64")
         expected = DataFrame({"A": ser})
         tm.assert_frame_equal(df, expected)
 
+    def test_loc_setitem_frame_with_reindex_mixed(self):
+        # same test as above, but with mixed dataframe
+        df = DataFrame(index=[3, 5, 4], columns=["A", "B"], dtype=float)
+        df["B"] = "string"
+        df.loc[[4, 3, 5], "A"] = np.array([1, 2, 3], dtype="int64")
+        # TODO with "split" path we still actually overwrite the column
+        # and therefore don't take the order of the indexer into account
+        ser = Series([1, 2, 3], index=[3, 5, 4], dtype="int64")
```

Comment:
this is weird right?  why wouldnt we expect `ser = Series([2, 3, 1], index=[3, 5, 4])`?

(actual question: is my intuition wrong, or is this a bug?)

Reply:
Yes, your intuition is correct, this is clearly a bug (that wasn't tested because the original test (still above) was using a homogeneous dataframe). That's what the TODO meant to explain. 
But I now opened an issue for it (https://github.com/pandas-dev/pandas/issues/40480, it's actually a regression on master, apparently), will update the comment with a link to that issue.

Reply:
And made it an xfail instead of testing the wrong result